common/scala/src/main/scala/org/apache/openwhisk/core/entity/Parameter.scala

```diff
@@ -80,15 +81,33 @@ protected[core] class Parameters protected[entity] (private val params: Map[Para
     params.keySet filter (params(_).init) map (_.name)
   }
 
+  protected[core] def getMap = {
+    params
```

Comment:
Yes, I need this in the ParamEncryption code so that the Parameter class doesn't try and do anything with it. I could move all that code here I suppose but I wanted to keep it logically separated.

Reply:
thanks for the explanation - i'm wondering if we can accomplish what you want by mixing in a trait - i'd have to look a bit more carefully so ignore my comment for now.